src/main/java/sparksoniq/jsoniq/runtime/iterator/RuntimeIterator.java

```diff
@@ -225,14 +226,14 @@ public IteratorMetadata getMetadata() {
             }
         }
         iterator.close();
-        if (result != null && !(type.isInstance(result)))
+        if (result != null && !(result.canBePromotedTo(new ItemType(ItemTypes.valueOf(type.getSimpleName())))))
```

Comment:
This method is used at many different places in the Rumble codebase. Promotion, however, only applies to specific places such as function calls, order by and some operators.

https://www.w3.org/TR/xquery-30/#id-type-promotion-and-operator-mapping

If you would like to do this change, it should be checked that all callers of this function correspond to these cases (many probably do). If this is not the case for some callers, then a boolean could be added as an extra parameter to allow returning an item that is only promotable to the desired type, or not.

https://www.w3.org/TR/xquery-30/#id-type-promotion-and-operator-mapping

Reply:
I realized that I do not really need this change as the promotion is happening before, thus the check with isInstance works already. 
_change reverted_